lighthouse-core/gather/driver.js

```diff
@@ -499,19 +500,45 @@ class Driver {
       };
 
       const domContentLoadedListener = () => {
+        hasDCLFired = true;
         if (networkStatusMonitor.is2Idle()) {
           onIdle();
         } else {
           onBusy();
         }
       };
 
+      // We frequently need to debug why LH is still waiting for the page.
+      // This listener is added to all network events to verbosely log what URLs we're waiting on.
+      const logStatus = () => {
+        if (!hasDCLFired) {
+          log.verbose('Driver', 'Waiting on DomContentLoaded');
+          return;
+        }
+
+        const inflightRecords = networkStatusMonitor.getInflightRecords();
```

Comment:
i've been a little concerned about the length of our `NetworkRecord` callbacks. I was seeing 1+ms for some while doing some OYB profiling a while back, and there are a lot of callbacks.

Can we detect whether or not we're in `verbose` mode before doing any of this work? Short of `settings` in here or changing lighthouse-logger again, I'm not sure there is, sadly.

Reply:
+1 to this and another lighthouse-logger change, I want one level in between verbose and default (debug or info perhaps?) that doesn't contain *every* protocol message

Reply:
Good call. Yah not showing events from Network domain would be my number one request.